tests/func-tests/defaults_test.go

```diff
@@ -249,4 +251,32 @@ var _ = Describe("Check Default values", Label("defaults"), Serial, func() {
 			Entry("when removing /spec", "/spec"),
 		)
 	})
+
+	Context("validation for", func() {
+		one := 1
+		two := 2
+		DescribeTable("AutoCPULimitNamespaceLabelSelector", func(allocationRatio *int, outcome gomegatypes.GomegaMatcher) {
+			restoreDefaults(cli)
+
+			hc := tests.GetHCO(ctx, cli)
+			hc.Spec.ResourceRequirements = &v1beta1.OperandResourceRequirements{
+				VmiCPUAllocationRatio: allocationRatio,
+				AutoCPULimitNamespaceLabelSelector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{"someLabel": "true"},
+				},
+			}
+			Eventually(func() error {
+				var err error
+				hc, err = tests.UpdateHCO(ctx, cli, hc)
+				if err != nil {
+					Expect(err.Error()).NotTo(ContainSubstring("the object has been modified; please apply your changes to the latest version and try again"))
```

Comment:
What is this check? Why do we need it?

Also, failed Expect will break the eventually loop.

Reply:
This is to ensure that we don't count update errors as the error we expect in the failure case.
Are you sure it breaks the loop? I thought it "continue"d it. Either way, I can change that to a `Consistently()` call for the failure case, that probably makes more sense.

Reply:
I think we don't care about the error. We need to wait  for no error. See other tests case in the same file

Reply:
Well I do want the update to fail when the ratio is 1. Anyway, will improve, thanks for the review!

Reply:
OK, after reading again, I understand your point. 

However:
1. I would move this test to another file. This test file is all about default values of the HyperConverged CR. I find it a bit confusing to check the CRD rule here. I'm not sure there is an existing better place to put it, so please consider to move this test to a new file.
2. As said above, using `Expect` within `Eventually`, break the loop in case of failure. There are several ways to address it. for example, using this pattern:
    ```go
    Eventually(func(g Gomega){
        ...
        g.Expect(...).To(...)
        ...
    }).To(...)
    ```
3. the result of `tests.UpdateHCO(ctx, cli, hc)` is assigned back to the `hc` variable, overriding its value (probably to nil, in case of error). So if we do get a conflict error and we'll want to retry, the retry will fail. I would use a new variable for that, defined outside of the Eventually loop, then we could check if the value was set correctly. We can also use `_` - that fine too. 
4. optional: maybe, instead of `Not(Succeed())`, we can use the `MatchError()` matcher? not sure if it will work. But if it will, we won't need the conflict error logic at all.
5. Please remove the `one` and `two` variables, and use `pointer.Int(1)` and `pointer.Int(2)` instead.

Reply:
Thanks a lot, all your points make sense, will update later today!